server/datastore/mysql/software_installers.go

```diff
@@ -237,7 +237,7 @@ func (ds *Datastore) SaveInstallerUpdates(ctx context.Context, payload *fleet.Up
 	}
 
 	var postInstallScriptID *uint
-	if payload.PostInstallScript != nil && *payload.PostInstallScript != "" { // pointer because optional
+	if payload.PostInstallScript != nil { // pointer because optional
```

Comment:
If the script is empty, we should set its ID to NULL in `software_installers`

Reply:
Yep, just realized that that was the previous behavior (postInstallScriptID *uint zero value being nil in L239). I've reverted the commit made based on your previous feedback. I had actually tested bouncing this back and forth and thought I had missed something when you brought the feedback up initially.

Reply:
Just verified this manually, going from blank post-install to populated, populated to different populated, and populated to blank. Values changed in the DB, including to/from NULL on post-install script contents ID.